File: iOS/App/ProductsCrud/Presentation/Product/Mapper/DomainToPresentationProductMapper.swift
//
//  DomainToPresentationProductMapper.swift
//  ProductsCrud
//
//  Created by Emil Falkowski on 12/11/2024.
//

struct DomainToPresentationProductMapper {
    static func map(model: Product) -> PresentationProduct {
        return PresentationProduct(
            id: model.id,
            title: model.title,
            description: model.description,
            price: model.price,
            category: model.category,
            images: model.images
        )
    }
}

File: Flutter/App/lib/presentation/product/mapper/domain_to_presentation_product_mapper.dart
import 'package:products_crud/domain/product/model/product.dart';
import 'package:products_crud/presentation/product/model/presentation_product.dart';

extension PresentationProductMapper on Product {
  PresentationProduct toPresentation() {
    return PresentationProduct(
      id: id,
      title: title,
      description: description,
      price: price,
      category: category,
      images: images,
    );
  }
}
